client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,76 @@
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import React, { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+import { useActions } from './use-actions';
+import { useFields } from './use-fields';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( getPluginActionStatuses );
+	const translate = useTranslate();
+
+	const fields = useFields( bulkActionDialog );
+	const actions = useActions( bulkActionDialog );
+
+	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( () => ( {
+		...initialDataViewsState,
+		perPage: 15,
+		search: initialSearch,
+		fields: [ 'plugins', 'sites', 'update' ],
+	} ) );
+
+	// When search changes, notify the parent component
+	useEffect( () => {
+		onSearch && onSearch( dataViewsState.search || '' );
+	}, [ dataViewsState.search, onSearch ] );
+
+	const { data, paginationInfo } = useMemo( () => {
+		const result = filterSortAndPaginate( currentPlugins, dataViewsState, fields );
+
+		// Add all statuses to the plugin object so we can display them in the DataViews
+		result.data.forEach( ( plugin ) => {
+			plugin.allStatuses = allStatuses.filter( ( status ) => status.pluginId === plugin.id );
+		} );
+
+		return {
+			data: result.data,
+			paginationInfo: result.paginationInfo,
+		};
+	}, [ currentPlugins, dataViewsState, fields, allStatuses ] );
+
+	return (
+		<DataViews
+			data={ data }
+			view={ dataViewsState }
+			onChangeView={ setDataViewsState }
+			fields={ fields }
+			search
+			searchLabel={ translate( 'Search for plugins' ) }
+			actions={ actions }
+			getItemId={ ( item ) => item.id }
+			isLoading={ isLoading }
+			paginationInfo={ paginationInfo }
+			defaultLayouts={ { table: {} } }
```

Comment:
This object should live on as a [module constant](https://github.com/WordPress/gutenberg/blob/80248a5dc082dd253d0e01b70174390fb8ca7646/packages/edit-site/src/components/page-patterns/index.js#L41C7-L41C13) or at least under some memoisation, otherwise the [memo work down the render tree](https://github.com/WordPress/gutenberg/blob/80248a5dc082dd253d0e01b70174390fb8ca7646/packages/dataviews/src/components/dataviews-view-config/index.tsx#L594) is invalidated.

Reply:
Can you confirm which object you are referring to, @mcsf? Is the `table` one?

Reply:
Sorry, I mean all of `defaultLayouts`, as done in the first link.

```suggestion
			defaultLayouts={ defaultLayouts }
```

Reply:
Can you confirm [if this is](https://github.com/Automattic/wp-calypso/pull/95561/commits/80809938bacfc28e3bcaa88d3e9619325c6a1770) what you mean?

Reply:
No, otherwise we're still creating a new object on every render. Of course, you could use `useMemo`, but this is what I was talking about:

```diff
diff --git a/client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx b/client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx
index 409e3d5956..84f0a94e14 100644
--- a/client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx
+++ b/client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx
@@ -18,6 +18,8 @@ interface Props {
 	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
 }
 
+const defaultLayouts = { table: {} };
+
 export default function PluginsListDataViews( {
 	currentPlugins,
 	initialSearch,
@@ -26,7 +28,6 @@ export default function PluginsListDataViews( {
 	bulkActionDialog,
 }: Props ) {
 	const translate = useTranslate();
-	const defaultLayouts = { table: {} };
 	const fields = useFields( bulkActionDialog );
 	const actions = useActions( bulkActionDialog );

```

Reply:
Oh, that's interesting. TIL. Can you share why it invalidates the `memo` if it's defined inside the module?
I [fixed it here](https://github.com/Automattic/wp-calypso/pull/95561/commits/927d3711a99ebdac655a86944375021063e244e7).

Reply:
> Can you share why it invalidates the memo if it's defined inside the module?

I probably explained poorly. I just wanted the `defaultLayouts` prop to be stable, i.e. not be a different object on every render of `PluginsListDataViews`. Defining it as a constant at the top of the file (what I referred to as module) is one way to do it, as found in [prior art](https://github.com/WordPress/gutenberg/blob/80248a5dc082dd253d0e01b70174390fb8ca7646/packages/edit-site/src/components/page-patterns/index.js#L41C7-L41C13). There's other ways to achieve the same thing, obviously, including 

```js
function PluginsListDataViews( ... ) {
  const defaultLayouts = useMemo( () => ( { table: {} } ), ...
```

But the important part was really just to keep the prop stable. If it keeps changing, then when `DataViews` re-renders a new value will be used in the child component `DataViewsViewConfig`. This component has been [optimised specifically](https://github.com/WordPress/gutenberg/blob/80248a5dc082dd253d0e01b70174390fb8ca7646/packages/dataviews/src/components/dataviews-view-config/index.tsx#L594) on the premise that its props aren't _constantly_ changing. So if `defaultLayouts` keeps changing that whole part of Data Views will be re-rendering needlessly. It's not a big deal, but just the right habit.

Reply:
Thanks so much for offering this extra context! Preventing it from re-rendering is critical for this project since we're preparing it to handling thousands of plugins!